### PR TITLE
chore: downgrade to nmt 0.12.0

### PR DESCRIPTION
**Pull request by rootulp**
## Description

Downgrade to nmt 0.12.0 until https://github.com/celestiaorg/celestia-node/pull/1651 is resolved

### CODE REVIEW

**Comment by evan-forbes** [comment]
we can do this in core and the sdk, or in my most recent PR, I simply added a replace state in the app. that way, we only have to change one thing to upgrade

**Comment by evan-forbes** [comment]
I'm fine with either approach though, and will approve once CI sets in

**Comment by rootulp** [comment]
> I'm fine with either approach though

Same. Since I'm cutting a new release of celestia-core, I didn't want it to use a dep that has a known issue so figured I would also downgrade here.

> and will approve once CI sets in

Thanks! FWIW, I think that if you approve early GH won't merge this PR until all required CI checks pass.

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**
